### DISCUSSION TITLE
How is nhibernate at handling postgres in general npgsql specifically

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [nhibernate/nhibernate-core#3595] [Q&A]
[_See issue_](https://github.com/npgsql/npgsql/issues/5797) dealing with an _Entity Framework Core_ issue, something between that and [`Npgsql`](https://nuget.org/packages/Npgsql) integration around a `pg` `JSON NOT NULL` column is failing horribly. Not sure how much better [`NHibernate`](https://nuget.org/packages/NHibernate) would be around that issue, is it an integration failure, or a driver failure, etc, for instance. Thoughts 🧠 ?

**Comment by fredericDelaporte** [comment]
NHibernate does not currently have support for a Json column type, so, "out of the box", that is unsupported. See [types](https://nhibernate.info/doc/nhibernate-reference/mapping.html#mapping-types). About how it would fail would have to be tested.

But NHibernate is quite extensible, and you can define [custom types](https://nhibernate.info/doc/nhibernate-reference/mapping.html#mapping-types-custom) for supporting about anything.
